stdlib/Random/src/XoshiroSimd.jl

```diff
@@ -0,0 +1,350 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+module XoshiroSimd
+# Getting the xoroshiro RNG to reliably vectorize is somewhat of a hassle without Simd.jl.
+import ..Random: TaskLocal, rand, rand!, UnsafeView, Xoshiro, SamplerType
+
+
+
+#Vector-width. Influences random stream. We may want to tune this before merging.
+xoshiroWidth() = Val(8)
+#Simd threshold. Influences random stream. We may want to tune this before merging.
+simdThreshold(::Type{T}) where T = 2048
+simdThreshold(::Type{Bool}) = 4096
+
+@inline _rotl45(x::UInt64) = (x<<45)|(x>>19)
+@inline _shl17(x::UInt64) = x<<17
+@inline _rotl23(x::UInt64) = (x<<23)|(x>>41)
+@inline _plus(x::UInt64,y::UInt64) = x+y
+@inline _xor(x::UInt64,y::UInt64) = xor(x,y)
+@inline _and(x::UInt64, y::UInt64) = x & y
+@inline _or(x::UInt64, y::UInt64) = x | y
+@inline _lshr(x, y::Int32) = _lshr(x, y % Int64)
+@inline _lshr(x::UInt64, y::Int64) = Core.Intrinsics.llvmcall("""
+    %res = lshr i64 %0, %1
+    ret i64 %res
+    """,
+    UInt64,
+    Tuple{UInt64, Int64},
+    x, y)
+
+
+
+# required operations. These could be written more concise with `ntuple`, but the compiler 
+# sometimes refuses to properly vectorize. 
+for N in [1,2,4,8,16,32]
+    let
+        local code
+        code = """
+        %lshiftOp = shufflevector <1 x i64> <i64 45>, <1 x i64> undef, <$N x i32> zeroinitializer
+        %rshiftOp = shufflevector <1 x i64> <i64 19>, <1 x i64> undef, <$N x i32> zeroinitializer
+        %lshifted = shl <$N x i64> %0, %lshiftOp
+        %rshifted = lshr <$N x i64> %0, %rshiftOp
+        %res = or <$N x i64> %lshifted, %rshifted
+        ret <$N x i64> %res
```

Comment:
You could use the [funnel shift](https://llvm.org/docs/LangRef.html#llvm-fshr-intrinsic) intrinsic.
Although LLVM compiles either approach into to
```asm
vprolq  $45, %zmm0, %zmm0
```
given AVX512.

Reply:
Afaiu fshr was introduced in llvm 7.0.0 (release late '18) -- do we still support older llvm versions? Regardless, either compiles ok at this moment.